### ISSUE TITLE
Aktivní POI: Slovensko ?

### ISSUE THREAD

**Issue by zbycz** [osmcz/osmcz#84]
Co takhle importnout ještě bratry slováky? Byla by to pro server větší zátěž? 

Možná jen POIky - tj nerenderovat tiles, atd.. což by se zase asi muselo nekonfigurovat.

**Comment by mkyral** [comment]
I když se nerenderují tiles, stejně by ta data pedro musel naimportovat, protože je json vrstva potřebuje. Určitě by to znamenalo docela významné navýšení požadavků na místo a výkon na serveru.

**Comment by Krakonos** [comment]
Asi bych to pro tuto chvíli nechal a raději soustředil síly na dobrou použitelnost pro ČR. Časem bych se spíš zamyslel, zda nenaimportovat pár km za hranicemi všech sousedních států, přecejenom se tam občas podíváme a mapy se hodí.

Nemají slováci nějaký vlastní podobný portál? Nějaký odkaz jsem viděl; případně bychom mohli spojit síly.

**Comment by mkyral** [comment]
Slováci mají freemap.sk.

Jinak souhlas. Zatím máme mezery jak v zobrazení těch POI tak i v rychlosti načítání. Když se to při zoomech seká (mazání a nové načítání vrstvy), tak to moc příjemné není. Nedá se to v tom javascriptu nějak prioritizovat, aby se aktivní vrstva načítala s nižší prioritou a neblokovala ostatní funkce, jako je načítání dlaždic a pohyb po mapě?

**Comment by zbycz** [comment]
Jo, s tím něco dělat jde. Chystám se na to ;)  Těch oblastí ke zlepšování je tu všude tolik :-)

Spíš mi šlo o to, že import SR může být otázkou jednoho řádku u pedra a třeba by se slováci přidali k vývoji a zlepšování :)

Jinak pár kilometrů za hranice je většinou problém, protože downloady jsou pro celé státy nebo pak až celou evropu :-/

**Comment by mkyral** [comment]
Tak čistě teoreticky to může být jeden řádek, ale s velkým dopadem na velikost databáze ;-)

**Comment by mkyral** [comment]
>  Těch oblastí ke zlepšování je tu všude tolik :-)

@zbycz Má smysl, abych se dále zabýval těmi POI? Jiná šablona pro restauraci/obchod, veřejnou instituci, rozcestník, ...? Nebo se mám raději mrknout na něco jiného?  Ale tady už jsme zase offtopic :-D

**Comment by mkyral** [comment]
> Jinak pár kilometrů za hranice je většinou problém, protože downloady jsou pro celé státy nebo pak až celou evropu :-/

Jednak jsem koukal, že pár metrů za hranice to většinou zasahuje ;-)

Jinak [Speirs](http://osm.paws.cz/) ve své `ee` mapě příhraničí generuje. Třeba by pomohl se skripty. Na talk-cz se občas vyskytuje.

Asi není potřeba stahovat komplet celou Evropu. Stačí jen příhraniční státy a z nich před importem vyříznout potřebné části.

**Comment by pedro042** [comment]
http://tile.poloha.net/jsonsk/...
testujte, pls. Když to bude fungovat, pak to spojím do jednoho.

**Comment by mkyral** [comment]
@pedro042 To Slovensko jsi tam už měl, nebo to ne nový import? A kolik dat navíc to představuje?

**Comment by zbycz** [comment]
Já jsem si říkal, že to slovensko bude pro pedra hozená rukavička :-)))

Testujte s ?active=sk:
http://rawgit.com/osmcz/osmcz/master/index.html?active=sk#map=13/48.1436/17.1398&layers=bA

**Comment by pedro042** [comment]
No, mně se to zdá OK. Fce jsou předělané tak, aby jim stačilo jen schema pro Mapnik a navíc jsou o 20-30% svižnější. Slováci by ke zprovoznění potřebovali Postgre + hstore, PostGIS a PHP (jelikož v tom PHP je asi 10 řádků, tak vlastně cokoli, co předá do Postgre x/y/z dlaždice a zpátky převezme jeden řetězec.
Slovensko je nové schema (sk), jen Mapnik schema, jak jsem už připodotkl. Sežralo to cca 4GB. Návrh těch schemat je v OSM neuvěřitelně neefektivní, spousta dat je tam vícekrát úplně zbytečně. Zřejmě z historických důvodů.
Jak jsem psal o tom spojování, tak se mi to už moc nelíbí. Nepříjemná je hlavně duplicita dat kolem hranic. Distinct pěkně zamává s prováděcími plány. Jakékoli řešení, které mě napadlo, by mělo negativní vliv na svižnost. Nevím, jak velký, ale proč to zpomalovat. Jestli Slováci mají Mapnik (jako že asi jo), tak pro ně nasazení musí být triviální. Nejraději bych se toho Slovenska na poloha.net zbavil, protože přeci jen fokus je na ČR a RÚIAN.

**Comment by zbycz** [comment]
Pěkný počin, zkusím jim napsat do konfery, jestli by to chtěli nějak převzít.

To spojování takto asi nepůjde, ovšem bavili jsme se někde, že by bylo zajímavé vysekat z europe.osm kromě čech i nějakých pár kiláků okolí, např pro využití na našich horách, pak by šlo uvažovat o zapojení i SK. Ale záleží asi na ohlasu..

**Comment by tjamecny** [comment]
Na freemap.sk už POI máme, len treba trochu viac klikať: napr. pre Olomouc - http://www.freemap.sk/#p=49.59154|17.25423|15|A v ľavom hornom rohu kliknúť na "Objekty", zvoliť si kategóriu a kliknúť na "Zobraziť POI". Nie je to úplne dokonalé, napr. po odzoomovaní sa automaticky nenačítajú POI, ale doteraz sa nikto nesťažoval :)

Na serveri používame PHP a MariaDB. V DB držíme jednak surové OSM dáta (zvlášť tabuľky pre node/way/relation, ďalej ich properties a members) a taktiež už spracované dáta (napr. zoznam POI, viditeľnoť názvov miest/dedín/lokalít tak, aby sa neprekrývali). Žiadny Mapnik/PostgreSQL/PostGIS nehrozí, možno až v ďalšej verzii Freemap.

Z osobných skúseností odporúčam stiahnuť europe.pbf (cca 16GB sa stiahne za 35 minút), a orezať na BBOX cez osmosis (30 minút). Potom už len stačí sťahovať denné diff-y (cca 65MB v GZ formáte).

**Comment by pedro042** [comment]
Škoda. To na poloha.net/openstreetmap.cz je brutálně svázané s Postgre, na Maria by asi nebylo triviální to navázat.
